backend/app/services/auth/utils.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime, timedelta
+
+import httpx
+from app.core.config import settings
+from app.services.auth.schema import TokenData
+from app.services.users.utils import add_user_to_users_json
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+from httpx import HTTPStatusError, Response
+from jose import JWTError, jwt
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+
+def create_jwt_token(*, data: dict, expires_delta: timedelta | None = None, token_type: str = "access") -> str:
+    to_encode = data.copy()
+    if expires_delta is None:
+        if token_type == "access":
+            expires_delta = settings.ACCESS_TOKEN_EXPIRE_MINUTES
+        elif token_type == "refresh":
+            expires_delta = settings.REFRESH_TOKEN_EXPIRE_DAYS
+        else:
+            raise ValueError(f"Unknown token type {token_type}")
+    expire = datetime.utcnow() + expires_delta
+    to_encode.update({"exp": expire})
+    to_encode["sub"] = str(to_encode["sub"])
+    encoded_jwt = jwt.encode(to_encode, settings.SECRET_KEY, algorithm=settings.ALGORITHM)
+    return encoded_jwt
+
+
+async def get_current_user(token: str = Depends(oauth2_scheme)) -> TokenData:
+    try:
+        payload = jwt.decode(token, settings.SECRET_KEY, settings.ALGORITHM)
+        github_id: str = payload.get("sub")
+        username: str = payload.get("username")
+        check_if_github_id_and_username_are_provided(github_id, username)
+        token_data = TokenData(github_id=github_id, username=username)
+    except JWTError:
+        raise get_credentials_exception()
+    return token_data
+
+
+def get_credentials_exception() -> HTTPException:
+    return HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
```

Comment:
This forces us to check what is the `get_credentials_exception` - creating separate file for exceptions and creating new one based on HTTPException would be more clear `class UnauthorizedException`